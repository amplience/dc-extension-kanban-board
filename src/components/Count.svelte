<script>
  export let count: number = 0;
  export let total: number = 0;
  export let additionalInfo: string | undefined;
</script>

<style lang="scss">
  .count {
    font-size: 13px;
    color: #333;
    margin: 10px;
  }
  .bold {
    color: #333;
    font-size: 16px;
    font-weight: 500;
  }
</style>

<div class="count">
  <span class="bold">{count}</span>
  content item{#if count !== 1}s{/if}
  {#if count !== total}of {total}{/if}
  {#if additionalInfo}{additionalInfo}{/if}
</div>
