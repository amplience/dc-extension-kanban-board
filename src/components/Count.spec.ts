import { render } from '@testing-library/svelte';
import Count from './Count.svelte';

describe('count component', () => {
  it('should render a count component with count and total value of 1', () => {
    const { container } = render(Count, {
      count: 1,
      total: 1,
      additionalInfo: 'ADDITIONAL_INFO',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a count component with count greater than 1', () => {
    const { container } = render(Count, {
      count: 2,
      total: 4,
      additionalInfo: 'ADDITIONAL_INFO',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a count component with default values', () => {
    const { container } = render(Count, {
      additionalInfo: '',
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
