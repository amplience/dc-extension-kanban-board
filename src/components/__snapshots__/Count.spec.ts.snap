// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`count component should render a count component with count and total value of 1 1`] = `
<div>
  <div
    class="count svelte-1aviz9u"
  >
    <span
      class="bold svelte-1aviz9u"
    >
      1
    </span>
    
  content item
     
     
    ADDITIONAL_INFO
  </div>
</div>
`;

exports[`count component should render a count component with count greater than 1 1`] = `
<div>
  <div
    class="count svelte-1aviz9u"
  >
    <span
      class="bold svelte-1aviz9u"
    >
      2
    </span>
    
  content item
    s
     
    of 
    4
     
    ADDITIONAL_INFO
  </div>
</div>
`;

exports[`count component should render a count component with default values 1`] = `
<div>
  <div
    class="count svelte-1aviz9u"
  >
    <span
      class="bold svelte-1aviz9u"
    >
      0
    </span>
    
  content item
    s
     
     
  </div>
</div>
`;
